### DISCUSSION TITLE
Visit promises deprecated

### DISCUSSION THREAD

**Discussion by liam-ot** [inertiajs/inertia#842] [Help] [answered]
Hello

This issue has sprung up for me this morning

```
index.js?5b3d:1 Inertia.js visit promises have been deprecated and will be removed in a future release. Please use the new visit event callbacks instead.
```

I have read the PR and the Inertia docs, I just don't understand what I'm doing wrong

here is the post request
```
method() {
    return this.$inertia.post(this.route('route_name', this.prop));
}
```

I don't understand what I am supposed to be calling instead of `this.$inertia.post()`

**Comment by xavi7th** [comment]
I had this problem for a bit with SweetAlert2 preconfirm requests and it buggered me.

The problem is the return. You shouldn't return inertia requests because they get converted to promises.

**Reply by liam-ot** [reply]
@xavi7th hmm, okay what exactly can i return then instead which will keep the benefits of XHR that inertia is built to take advantage of? 

thanks for the reply

**Reply by xavi7th** [reply]
@liam-ot well I can't particularly see your logic so I don't know what you want to achieve with the return, but without the return, the inertia request will work just fine.

**Comment by liam-ot** [comment, marked as answer]
the issue as mentioned by @xavi7th is due to the return type being turned into a promise

if you are routing like this
```
Redirect::route('someRoute')->with('success', $success_message);
```
or
```
Redirect::back()->with('success', $success_message);
```

change it instead to
```
return redirect()->back()->with('success', $success_message);
```
 unfortunately this still causes a promise warning
```
return redirect()->route('someRoute')->with('success', $success_message);
```
but i will update once i find a workaround
